applications/luci-app-sshtunnel/htdocs/luci-static/resources/view/sshtunnel.js

```diff
@@ -0,0 +1,403 @@
+'use strict';
+'require form';
+'require fs';
+'require uci';
+'require ui';
+'require view';
+
+var allSshKeys = {};
+var hasSshKeygen = false;
+
+return view.extend({
+	load: function () {
+		return L.resolveDefault(fs.list("/root/.ssh/"), []).then(function (entries) {
+			var tasks = [
+				uci.load('sshtunnel'),
+				L.resolveDefault(fs.stat("/usr/bin/ssh-keygen"), {}),
+				fs.lines("/root/.ssh/known_hosts"),
+			];
+			// read pub keys
+			for (var i = 0; i < entries.length; i++) {
+				if (entries[i].type === "file" && entries[i].name.match(/\.pub$/)) {
+					tasks.push(Promise.resolve(entries[i].name));
+					tasks.push(fs.lines("/root/.ssh/" + entries[i].name));
+				}
+			}
+			return Promise.all(tasks);
+		});
+	},
+
+	render: function (sshFiles) {
+		hasSshKeygen = sshFiles[1].type === "file";
+		var knownHosts = sshFiles[2];
+		var sshKeys = _splitSshKeys(sshFiles.splice(3));
+		if (sshKeys.length === 0) {
+			ui.addNotification(null, E("p", [
+				_("No any SSH key, please generate a new one")
+			]), "warning");
+		}
+
+		var m, s, o;
+
+		m = new form.Map("sshtunnel", _("SSH Tunnels"),
+			_("This configures <a href='https://openwrt.org/docs/guide-user/services/ssh/sshtunnel'>SSH Tunnels</a>. " +
+				"Password auth is not possible so only Public Key auth must be used.")
+		);
+
+		s = m.section(form.GridSection, "server", _("Servers"));
+		s.anonymous = false;
+		s.addremove = true;
+		s.nodescriptions = true;
+
+		o = s.tab("general", _("General Settings"));
+		o = s.tab("advanced", _("Advanced Settings"));
+
+		o = s.taboption("general", form.Value, "hostname", _("Hostname"), "");
+		o.placeholder = "example.com";
+		o.datatype = "host";
+		o.rmempty = false;
+
+		o = s.taboption("general", form.Value, "port", _("Port"), "");
+		o.placeholder = "22";
+		o.datatype = "port";
+
+		o = s.taboption("general", form.Value, "user", _("User"), "");
+		o.default = "root";
+
+		o = s.taboption("general", form.ListValue, "IdentityFile", _("Key file"),
+			_("Private key file with authentication identity. The default is /root/.ssh/id_ed25519, id_rsa, id_ecdsa or id_dropbear.")
+		);
+		o.value("", "");
+		Object.keys(sshKeys).forEach(function (keyName) {
+			o.value("/root/.ssh/" + keyName, keyName);
+		});
+		o.optional = true;
+
+		o = s.taboption("advanced", form.ListValue, "CheckHostIP", _("Check host IP"),
+			_("CheckHostIP If enabled the ssh will check the host IP address in the known_hosts file. " +
+				"This allows ssh to detect if a host key changed due to DNS spoofing.")
+		);
+		o.value("yes", _("Yes"));
+		o.value("no", _("No"));
+		o.default = "no";
+		o.modalonly = true;
+
+		o = s.taboption("advanced", form.ListValue, "VerifyHostKeyDNS", _("Verify Host Key DNS"),
+			_("VerifyHostKeyDNS Verify the remote key using DNS and SSHFP resource records.")
```

Comment:
I removed the VerifyHostKeyDNS option from GUI. It's highly unlikely that it will be used and it's anyway disabled by default. If anyone wants to use it they still can do this with UCI.